### DISCUSSION TITLE
How to record audio streams on a server and save as flv file for FLV Debugger?

### DISCUSSION THREAD

**Discussion by wilson-pok** [airsdk/Adobe-Runtime-Support#2041] [Q&A] [answered]
Currently, I am using "FLV Debugger" tool to record a digital radio stream for debugging metadata cue point accuracy.
![image](https://user-images.githubusercontent.com/32567675/182117830-6f3be896-d16f-43e7-b00a-6f96997e635a.png)

Using the tool is very manual. Is there a way I can develop a java-based application to record the digital radio stream every hour and save them into .flv files that can be opened by "FLV Debugger"? 
![image](https://user-images.githubusercontent.com/32567675/182118802-11bc67cd-ec60-4909-ab05-e07fd2dfafc5.png)

This way, the monitoring can automated for more than 1 radio streams.

**Comment by ajwfrost** [comment, marked as answer]
Hi

If the URLs that you're monitoring there really are just FLV streams, then I don't see why not. They presumably would give you an FLV file via chunked encoding so that it's a continuous stream, so as long as you get the header to start with, you can then just grab the FLV data and save it for later inspection. If you want to save them in chunks then you just need to add the headers onto each section..

FLV specification is here:
http://download.macromedia.com/f4v/video_file_format_spec_v10_1.pdf
More info may be available from Veriskope - the above document is also hosted on their website, the bottom link on this page:
https://rtmp.veriskope.com/docs/

thanks

   Andrew

**Reply by wilson-pok** [reply]
Thanks for your help. 

Do you have any java code sample to do this with a given URL that is flv continuous stream?
1) Get the header to start with
2) Grab the FLV data
3) Save it

**Reply by ajwfrost** [reply]
No, sorry, I don't think we've ever done this sort of thing with Java. Closest we've got to this is reading an MP4 stream and pulling out the individual a/v packets and then putting them into an FLV container to send into the Flash Player via `NetStream.appendBytes()` which was all written in C++, and a fair few years ago now. I would hope it's possible to read from a Java `HttpURLConnection` input stream and for that to keep providing the data - there's mention of chunked encoding there:
https://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#setChunkedStreamingMode(int)
So it may just be a case of checking what you get, and then adding a parser to handle the FLV headers vs the data..?
